### DISCUSSION TITLE
Incompatible with vcpkg versions

### DISCUSSION THREAD

**Discussion by acgetchell** [community/community#25243] [Actions]
<p>Short version:</p>
<pre><code class="lang-auto">Run vcpkg install --feature-flags=manifests
7
Error: while checking out port fmt with git tree dd8cf5e1a2dce2680189a0744102d4b0f1cfb8b6
8
Error: Failed to tar port directory
9
fatal: not a tree object: dd8cf5e1a2dce2680189a0744102d4b0f1cfb8b6
10

11
Error: while checking out port spdlog with git tree a2f8d7856c8821fb685f99873f5e058dc6136c2b
12
Error: Failed to tar port directory
13
fatal: not a tree object: a2f8d7856c8821fb685f99873f5e058dc6136c2b
14

15
Error: Process completed with exit code 1.
</code></pre>
<aside class="onebox allowlistedgeneric" data-onebox-src="https://github.com/acgetchell/CDT-plusplus/runs/3853321615">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">

      <a href="https://github.com/acgetchell/CDT-plusplus/runs/3853321615" target="_blank" rel="noopener nofollow ugc">GitHub</a>
  </header>

  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/434849/181089820-689f53cf-e7d8-4b26-b4a9-4a69a77e91d4.png" class="thumbnail onebox-avatar" width="500" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_500x500.png, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_750x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_1000x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_10x10.png">

<h3><a href="https://github.com/acgetchell/CDT-plusplus/runs/3853321615" target="_blank" rel="noopener nofollow ugc">Build software better, together</a></h3>

  <p>GitHub is where people build software. More than 65 million people use GitHub to discover, fork, and contribute to over 200 million projects.</p>

  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p>Long version:</p>
<p>I need to use version 7.1.3 of the fmt library, because the latest version doesn’t link on Windows.</p>
<p>The mechanism to do that is to specify an override:</p>
<aside class="onebox githubblob" data-onebox-src="https://github.com/acgetchell/CDT-plusplus/blob/603ff82bf27d848788de8c10339bb04253a21d81/vcpkg.json">
  <header class="source">

      <a href="https://github.com/acgetchell/CDT-plusplus/blob/603ff82bf27d848788de8c10339bb04253a21d81/vcpkg.json" target="_blank" rel="noopener nofollow ugc">github.com</a>
  </header>

  <article class="onebox-body">
    <h4><a href="https://github.com/acgetchell/CDT-plusplus/blob/603ff82bf27d848788de8c10339bb04253a21d81/vcpkg.json" target="_blank" rel="noopener nofollow ugc">acgetchell/CDT-plusplus/blob/603ff82bf27d848788de8c10339bb04253a21d81/vcpkg.json</a></h4>

    <pre><code class="lang-json">{
  "name": "cdt-plusplus",
  "version": "0.1.8",
  "dependencies": [
    "catch2",
    "docopt",
    "date",
    "fmt",
    "ms-gsl",
    "eigen3",
    "pcg",
    "spdlog",
    "tbb",
    "tl-expected",
    "tl-function-ref",
    {
      "name": "yasm-tool",
      "platform": "windows"
    },
    "cgal"
</code></pre>

  This file has been truncated. <a href="https://github.com/acgetchell/CDT-plusplus/blob/603ff82bf27d848788de8c10339bb04253a21d81/vcpkg.json" target="_blank" rel="noopener nofollow ugc">show original</a>

  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p>However, this doesn’t work on GitHub Actions and generates the error above. Note that it works perfectly fine on Travis-CI and AppVeyor:</p>
<p><a href="https://app.travis-ci.com/github/acgetchell/CDT-plusplus" class="onebox" target="_blank" rel="noopener nofollow ugc">https://app.travis-ci.com/github/acgetchell/CDT-plusplus</a></p>
<p><a href="https://ci.appveyor.com/project/acgetchell/cdt-plusplus" class="onebox" target="_blank" rel="noopener nofollow ugc">https://ci.appveyor.com/project/acgetchell/cdt-plusplus</a></p>
<p>Using the default version of the library in <code>vcpkg</code> works.</p>

**Comment by VanessaHuff** [comment]
<p>May be This Is Helpful For You <a href="https://github.com/actions/virtual-environments/issues/2127" rel="noopener nofollow ugc">https://github.com/actions/virtual-environments/issues/2127</a><a href="https://floridaloteryresults.com/" rel="noopener nofollow ugc">-results</a>.</p>

**Comment by ras0219-msft** [comment, marked as answer]
<p>I’ve opened <a href="https://github.com/actions/virtual-environments/pull/4272" class="inline-onebox" rel="noopener nofollow ugc">Perform a full clone of vcpkg instead of depth=1 by ras0219-msft · Pull Request #4272 · actions/virtual-environments · GitHub</a> which should address this issue.</p>
